src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
+bool SliproadHandler::isThroughStreet(const EdgeID from, const Intersection &intersection) const
+{
+    BOOST_ASSERT(from != SPECIAL_EDGEID);
+    BOOST_ASSERT(!intersection.empty());
+
+    const auto edge_name_id = node_based_graph.GetEdgeData(from).name_id;
+
+    auto first = begin(intersection) + 1; // Skip UTurn road
+    auto last = end(intersection);
+
+    auto same_name = [&](const auto &road) {
+        const auto road_name_id = node_based_graph.GetEdgeData(road.eid).name_id;
+
+        return edge_name_id != EMPTY_NAMEID && //
+               road_name_id != EMPTY_NAMEID && //
+               !util::guidance::requiresNameAnnounced(edge_name_id,
+                                                      road_name_id,
+                                                      name_table,
+                                                      street_name_suffix_table); //
+    };
+
+    return std::find_if(first, last, same_name) != last;
+}
+
+bool SliproadHandler::roadContinues(const EdgeID current, const EdgeID next) const
+{
+    const auto current_data = node_based_graph.GetEdgeData(current);
+    const auto next_data = node_based_graph.GetEdgeData(next);
+
+    auto same_road_category = current_data.road_classification == next_data.road_classification;
+    auto same_travel_mode = current_data.travel_mode == next_data.travel_mode;
+
+    auto same_name = current_data.name_id != EMPTY_NAMEID && //
+                     next_data.name_id != EMPTY_NAMEID &&    //
+                     !util::guidance::requiresNameAnnounced(current_data.name_id,
+                                                            next_data.name_id,
+                                                            name_table,
+                                                            street_name_suffix_table); //
+
+    const auto continues = same_road_category && same_travel_mode && same_name;
+    return continues;
+}
+
+bool SliproadHandler::isValidSliproadArea(const double max_area,
+                                          const NodeID a,
+                                          const NodeID b,
+                                          const NodeID c) const
+
+{
+    using namespace util::coordinate_calculation;
+
+    const auto first = node_info_list[a];
+    const auto second = node_info_list[b];
+    const auto third = node_info_list[c];
+
+    const auto length = haversineDistance(first, second);
+    const auto heigth = haversineDistance(second, third);
+
+    const auto area = (length * heigth) / 2.;
+
+    // Everything below is data issue - there are some weird situations where
+    // nodes are really close to each other and / or tagging ist just plain off.
+    const constexpr auto MIN_SLIPROAD_AREA = 3.;
+
+    if (area < MIN_SLIPROAD_AREA || area > max_area)
+        return false;
+
+    return true;
+}
+
+bool SliproadHandler::isValidSliproadLink(const ConnectedRoad &sliproad,
+                                          const ConnectedRoad &first,
```

Comment:
first/second don't really provide an idea of what other roads you are passing in.